<a href="https://colab.research.google.com/github/ManJ-PC/Py/blob/master/CursoEmV%C3%ADdeoMundo2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# if carro.esquerda():
#   bloco 1
# elif carro.direita():
#   bloco 2
# elif carro.ré():
#   bloco 3
# else:
#   bloco 4 NO

In [ ]:
# nome = str(input('Qual é o seu nome? '))
# if nome == 'Gustavo':
#   print('Que nome bonito!')
# elif nome in 'Ana Claúdia Jéssica Juliana':
#   print('Belo nome feminino')
# else:
#   print('Seu nome é bem normal.')
# print('Tenha um bom dia, {}!'.format(nome))

**Desafio  036**

Escreva um programa para aprovar o empréstimo bancário para a compra de uma casa. O Programa vai perguntar o valor da casa, o salário do comprador e em quantos anos ele vai pagar.

Calcule o valor da prestação mensal, sabendo que ela não pode exceder 30% do salário ou então o empréstimo será negado.

In [ ]:
# patrimonyL = float(input('Good Morning to our House Loan\'s Bank!\n Which the value of property that wants to adquire: '))
# wage = float(input('What is your wage: '))
# ageY = float(input('How much time lasts bank loan, in years: '))

# monthPay = patrimonyL/(ageY * 12)

# print('The mensal value to pay the house is: ', monthPay)

# if monthPay > 0.30 * wage:
#   print("Empréstimo negado")
# else:
#   print("Aceite!!")


In [ ]:
# #SOLUÇÂO

# casa = float(input('Valor da casa: R$  '))
# salario = float(input('Salário do comprador: R$  '))
# anos = int(input('Quantos anos de financiamento?  '))
# prestação = casa / (anos * 12)
# minimo = salario * 30 / 100 # min = .3*wage
# print('Para pagar uma casa de R${:.2f} em {} anos'.format(casa, anos), end='')
# """, end='')"""
# print(' a prestação será de R${:.2f}'.format(prestação))
# #print('Para pagar uma casa de R${:.2f}'.format(prestação))

# #print('COMPARANDO tem que pagar {} e o mínimo é de {}'.format(prestação,minimo))

# if prestação <= minimo: 
#    print('Empréstimo pode ser CONCEDIDO')
# else:
#   print('Empréstimo NEGADO')

**Desafio 37**

Escreva um programa que leia um número inteiro qualquer e peça para o usuário escolher qual será a **base de conversão**


*   1 para binário
*   2 para octal
*   3 para hexadecimal



In [ ]:
# num = int(input('Insira o valor decimal: '))
# b2 = bin(num)
# o8 = oct(num)
# Ox = hex(num)

# base = str(input('Qual será a base de conversão?\n O <- octal\n H <- hexadecimal\n B or other thing <- binary '))
# if base.upper() == 'O':
#   print(o8)
# elif base.upper() == 'H':
#   print(Ox)
# else:
#   print(b2)

In [ ]:
# SOLUÇÃO


# num = int(input('Digite um numero inteiro: '))
# print('''Escolha uma das bases para conversão:
# [ 1 ] converter para BINÁRIO
# [ 2 ] converter para OCTAL
# [ 3 ] converter para HEXADECIMAL''')
# opção = int(input('Sua opção: '))
# if opção == 1:
#   print('{} convertido para BINÁRIO é igual a {}'.format(num, bin(num)))
# elif opção == 2:
#   print('{} convertido para OCTAL é igual a {}'.format(num, oct(num)))
# elif opção == 3:
#   print('{} convertido para HEXADECIMAL é igual a {}'.format(num, hex(num)))

In [ ]:
# def converte2(a,b): #a = bq + r
#   novabase = "" #resto ou r_string
#   r = 0
#   while a>0: #r<a: #r > 0: ...n//b!
#     q = a // b
#     r = a - (q*b)
#     #r = a % b
#     a = q 

#     print(r,'\n')
#     #novabase=str(r)
#     novabase += str(r)
#   novabase = novabase[::-1]
#   #novabase[::-1]
#   #print('\n',novabase[::-1])
#   return novabase

In [ ]:
# def converte(n,b): #n = bq + r
#   # 10b10 = 1010b2  10/2 r1 5/2 r1 2/2 r0 0/2 r0
#   r_string = ""
#   while n//b > 0:
#     r = n%b
#     a = n//b
#     print(r,'\n')
#     r_string += str(r)
#   r_string[::-1]
#   print('')  NO
    

    


In [ ]:
# converte2(12,2)

In [ ]:
# converte2(12, 8)

In [ ]:
# Solution
def dec_to_base(num,base): # Maximum base - 36
  base_num = ""
  while num>0:
    dig = int(num%base) # porque este int? 
    if dig<10:
      base_num += str(dig)
    else:
      base_num += chr(ord('A')+dig-10) # Using uppercase Letters
    num //= base

  base_num = base_num[::-1] #To reverse the string
  return base_num

In [ ]:
dec_to_base(77,2)

In [ ]:
int("12",8)

In [ ]:
ord('a')

In [ ]:
ord('A')

In [ ]:
chr(94)+chr(93)+chr(92)+chr(91)+chr(90)

In [ ]:
# Any Base to Decimal

print(int("12",3))
print(int("12",8))
print(int("12",10))
print(int("12",16))

#chr(16)

In [ ]:
username = input("Choose a username: ")  # Canita Broa
final_username = username.replace("r", "_")

print("Your username is: " + final_username) 

In [ ]:
for element in final_username: #string_name:
  print(element, end=' ')
print("\n")

G a l o 



In [ ]:
def base_to_dec(num_str, base):
 base_num = 0
 power = 0           # declaration and initialization
 for column in num_str[::-1]: # d
   print(column)
   base_num += int(column) * pow(base, power)  # c = a + b 
   power += 1 # e
 print(base_num)
 #falta a potência!!! Incorporar, dessa forma em vez 12*8 = 96
# We have:

In [ ]:
num_str = "147"
num_str[::-1]

'741'

In [ ]:
base_to_dec("147",8)

7
4
1
103


In [ ]:
base_to_dec("147",8)

7
4
1
103


In [ ]:
print(pow(8,3))
7*64 # valores muito próximos, logicamente... pois base = 8 , 7 imediamente antes!

512


448

In [ ]:
base_to_dec("147",8)

7
4
1
103


In [ ]:
pow(4,4)

256

In [ ]:
pow(4,4,8)

0

In [ ]:
pow(4,4,12)

4

In [ ]:
# a,b,c = 100
# print(c)  .> cannot unpack non-iterable int object

a,b,c = 1,2,3
print(a,b,c)

1 2 3


In [ ]:
def baseToDec(num_str, base): # suporta até base de valor 36 
  num_str = num_str[::-1]
  num = 0
  for k in range(len(num_str)):
    dig = num_str[k]
    if dig.isdigit():
      dig = int(dig)
    else:  # Assuming its either number or alphabet only
      dig = ord(dig.upper())-ord('A')+10
    num += dig*(base**k)
  return num

In [ ]:
baseToDec("147",8)

103

In [ ]:
tries = 5
time = "0"
names = ""
while time:
  tries -= 1
  name = input('Digite seu nome: ')
  if name == 'name':
    break
  names += name
print(names)

Digite seu nome: iu
Digite seu nome: j
Digite seu nome: name
iuj


**Desafio 038**

Escreva um programa que leia dois números inteiros e compare-os, mostrando na tela uma mensagem:

- O primeiro valor é maior
- O segundo valor é maior 
- Não existe valor maior, os dois são iguais

In [ ]:
n1 = input('Digite o 1º: ')
n2 = input('Digite o 2º: ')
if n1 > n2:
  print('O primeiro valor é maior.')
elif n1 == n2:
  print('\nR: Não existe valor maior, os dois são iguais.')
else:
  print('\nR: O segundo valor é maior')

Digite o 1º: 6
Digite o 2º: 9

R: O segundo valor é maior


**Desafio 039**

Faça um programa que leia o ano de nascimento de um jovem e informe, de acordo com sua idade:

- Se ele ainda vai se alistar ao serviço militar.
- Se é na hora de se alistar.
- Se já passou do tempo de alistamento.

Seu programa também deverá mostrar o tempo que falta ou que passou de prazo

In [ ]:
from datetime import date

year = date.today().year
#print(data_atual)
bornYear = int(input('Say your born year: '))

MILITARYAGE = 18 
age = year-bornYear

militaryTime = age - MILITARYAGE

if militaryTime > 0:
  print('Already passed ',militaryTime, ' year(s) since military register' )
elif militaryTime == 0:
  print('Military register is in this year')
else:
  print(abs(militaryTime), 'year(s) to you have military service register day.')

Say your born year: 2023
19 year(s) to you have military service register day.


In [ ]:
data_atual = date.today()
data_em_texto = '{}/{}/{}'.format(data_atual.day, data_atual.month, data_atual.year)

In [ ]:
data_em_texto

'2/3/2022'

In [ ]:
import datetime

In [ ]:
data_e_hora_atuais = datetime.datetime.now() # truque
data_e_hora_em_texto = data_e_hora_atuais.strftime('%d/%m/%Y')

print(data_e_hora_em_texto)

02/03/2022


In [ ]:
data_e_hora_em_texto = data_e_hora_atuais.strftime('%d/%m/%Y %H:%M')

print(data_e_hora_em_texto)

02/03/2022 16:05


In [ ]:
from datetime import datetime

data_e_hora_em_texto = '17/07/2030 12:55' #14:61'
data_e_hora = datetime.strptime(data_em_texto, '%d/%m/%Y')# %H:%M')
data_e_hora = datetime.strptime(data_e_hora_em_texto, '%d/%m/%Y %H:%M')


In [ ]:
print(data_e_hora)

2030-07-17 12:55:00


**Desafio 40**

Crie um programa que leia duas notas de um aluno e calcule sua média, mostrando uma mensagem no final, de acordo com a média atingida:
- Média abaixo de 5.0:  REPROVADO

- Média entre 5.0 e 6.9: RECUPERAÇÃO

- Média 7.0 ou superior: APROVADO


In [ ]:
grade1 = float(input('Tell your first grade: '))
grade2 = float(input('Tell me your second grade: '))

avg = round((grade1+grade2)/2,1) # se 6.0 e 3.9 dá RECUPERAÇÃO

if avg < 5.0:
  print('REPROVADO')
elif avg < 7.0: 
  print('RECUPERAÇÃO')
else:
  print('APROVADO')
  

Tell your first grade: 7
Tell me your second grade: 9
APROVADO


**Desafio 41**

A Confederação Nacional de Natação precisa de um programa que leia o ano de nascimento de um atleta e mostre sua categoria, de acordo com a idade:

- Até 9 anos: MIRIM
- Até 14 anos: INFANTIL
- Até 19 anos: JUNIOR
- Até 30 anos: SÊNIOR
- Acima: MASTER

In [ ]:
year

2022

In [ ]:
bornYear = int(input('Athlete, say your born year: '))
age = year - bornYear
if age <= 9:
  print('MIRIM')
elif age <= 14:
  print('INFANTIL')
elif age <= 19:
  print('JUNIOR')
elif age <= 30:
  print('SÊNIOR')
else:
  print('MASTER')

Athlete, say your born year: 2018
MIRIM


**DESAFIO 42**

Refaça o Desafio 35 dos triângulos, acrescentando o recurso de mostrar que tipo de triângulo será formado:

* **Equilátero**: todos os lados iguais
* **Isósceles**: dois lados iguais
* **Escaleno**: todos os lados diferentes

In [ ]:
def istriangle(a,b,c):
  # great = a
  # if great<b:
  #   great = b
  # if great < c:
  #   great = c
  # #return great
  # great
  return a < b + c and b < a + c and c < a + b
 

In [ ]:
def triangletype(s1,s2,s3):
  if istriangle(s1,s2,s3):
    if s1 == s2 and s2 == s3:
      print('Equilateral')
    elif s1 == s2 or s1 == s3 or s2 == s3:
      print('Isosceles')
    else:
      print('Scalene')
  else: print('With this measures don\'t is possible build a triangle')

In [ ]:
triangletype(2,3,4)

Scalene


In [ ]:
triangletype(4,3,10)

With this measures don't is possible build a triangle


In [ ]:
def triangletype2(s1,s2,s3):
  if istriangle(s1,s2,s3):
    if s1 == s2:
      if s1 == s3:
        print('Equilateral')
      else: print('Isosceles')
    elif s1 == s3 or s2 == s3:
      print('Isosceles')
    else: print('Scalene') 
  else: print('With this measures don\'t is possible build a triangle')

In [ ]:
triangletype2(1,2,3)

With this measures don't is possible build a triangle


In [ ]:
triangletype2(4,2,5)

Scalene


In [ ]:
triangletype2(3,3,5)

Isosceles


In [ ]:
triangletype(0,0,0)

Equilateral


In [ ]:


#def istriangle(a,b,c): # decomponho para ver qual é o maior e a partir daí vejo se é triangulo
  #if (a>b>c)

In [ ]:
istriangle(2,2,3)

True

In [ ]:
# def istriangle(a,b,c):
#   great(a,b,c)# a > b + c
  

In [ ]:
#greaterSide(3,4,6)

6

**Desafio 043**

Desenvolva uma lógica que leia o peso e a altura de uma pessoa. Calcule seu **IMC** e mostre seu status, de acordo com a tabela abaixo:
- Abaixo de **18.5**: Abaixo do Peso
- Entre **18.5** e **25**: Peso ideal
- **25** até **30**: Sobrepeso
- Acima de **40**:
Obesidade mórbida

In [ ]:
DOWN = 18.5
RIGHT = 25
OVER = 30
OPUP = 40

altura = float(input('Digite a sua altura (em metros): '))
peso = float(input('Digite o seu peso (em kilogramas): '))



IMC = peso/altura**2

if IMC < DOWN:
  print('Below weight\'s')
elif IMC < RIGHT:
  print('Weight optimal ')
elif IMC < OVER:
  print('Above weight\'s')
elif IMC < OPUP:
  print('Morbid obesity')

Digite a sua altura (em metros): 1.89
Digite o seu peso (em kilogramas): 80
Weight optimal 


In [ ]:
IMC

22.395789591556788

**Desafio 044**
Elabore um programa que calcule o valor a ser pago por um produto, considerando o seu **preço normal** e **condição de pagamento**:


*   à vista **dinheiro/cheque: 10%** de desconto
*   à vista no **cartão: 5%** de desconto
*   em até **2* no cartão**: preço normal
*   **3* ou mais** no cartão: [**20%** de juros



In [ ]:
price = float(input('What is the price of your product: '))
payC = str(input('What is payment condition that you choose: ')) # paymentCOndition

if payC == '1':
  price *= 0.95
elif payC == '2':
  price *= 1
elif payC == '3':
  price *= 1.20
elif payC.lower == 'm' or 'c':
  price *= 0.9


print('O preço a ser pago mediante a condição de pagamento é', price)

What is the price of your product: 12
What is payment condition that you choose: c
O preço a ser pago mediante a condição de pagamento é 10.8


**Desafio 045**
Crie um programa que faça o computador **Jokenpô** com voçê.

In [ ]:
import random


In [ ]:
presentation = print('Rock, Paper and Scissors Game')

Rock, Paper and Scissors Game


In [ ]:
gameV =  {1:'rock', 2:'paper', 3:'scissor'}

while True: 
  num = random.randint(0,2) + 1
  human = int(input('Write a number between 1 to 3 which:\n 1 is Rock \n 2 is Paper \n 3 is Scissor\n'))
  if (0 < human < 4):
    break

print(num)
pc = gameV[num]
print(pc)
print(' Jokenpô GAME :\n\n Human move: ', gameV[human],' vs PC move:', pc)#, '\n \n RESULT: ')
dif = human-num #abs(num-human)

print(dif)
if dif == 0:
  print('Draw', pc)
elif dif == 1 or dif == -2:
  print('Vitory')
else:
  print('Defeat') 

#human = gameV[human]

#if 


Write a number between 1 to 3 which:
 1 is Rock 
 2 is Paper 
 3 is Scissor
4
Write a number between 1 to 3 which:
 1 is Rock 
 2 is Paper 
 3 is Scissor
8
Write a number between 1 to 3 which:
 1 is Rock 
 2 is Paper 
 3 is Scissor
-1


In [ ]:
gameV[2]

'paper'

In [ ]:
type(gameV[num])

str

In [ ]:
type(gameV)

dict

In [ ]:
#type(gameV[])

SyntaxError: ignored

In [ ]:
gameV[num]

'scissor'

In [ ]:
# Solução do Gustavo Guanabará
from random import randint
from time import sleep
itens =  ('Pedra', 'Papel', 'Tesoura')
computador = randint(0, 2)
print('''Suas opções:
[ 0 ] PEDRA
[ 1 ] PAPEL
[ 2 ] TESOURA''')
jogador = int(input('Qual é a sua jogada? '))
print('JO')
sleep(1)
print('KEN')
sleep(1)
print('PÔ!!!')
print('-=' * 11)
print('O computador escolheu {}'.format(itens[computador]))
print('Jogador jogou {}'.format(itens[jogador]))
print('-=' * 11)
if computador == 0: # computador jogou P  edra
  if jogador == 0:
    print('EMPATE')
  elif jogador == 1:
    print('JOGADOR VENCE')
  elif jogador == 2:
    print('COMPUTADOR VENCE')
  else:
    print('JOGADA INVÁLIDA!')
elif computador == 1: # computador jogou PAPEL
  if jogador == 0:
    print('COMPUTADOR VENCE')
  elif jogador == 1:
    print('EMPATE')
  elif jogador == 2:
    print('JOGADOR VENCE')
  else:
    print('JOGADA INVÁLIDA!')
elif computador == 1: # computador jogou TESOURA
  if jogador == 0:
    print('JOGADOR VENCE')
  elif jogador == 1:
    print('COMPUTADOR VENCE')
  elif jogador == 2:
    print('EMPATE')
  else:
    print('JOGADA INVÁLIDA!')

Suas opções:
[ 0 ] PEDRA
[ 1 ] PAPEL
[ 2 ] TESOURA
Qual é a sua jogada? 2
-=-=-=-=-=-=-=-=-=-=-=
O computador escolheu Papel
Jogador jogou Tesoura
-=-=-=-=-=-=-=-=-=-=-=
JOGADOR VENCE


In [ ]:
# Estrutura de repetição

# Aula 13

# print('Oi')
# print('Oi')
# print('Oi')
# print('Oi')
# print('Oi')
# print('Oi')


for c in range(0, 6):
  print('Oi')
print('FIM')

# aninhar, colocar uma estrutura dentro da outra

# sintaxe do portinhol e do ...

Oi
Oi
Oi
Oi
Oi
Oi
FIM


In [ ]:
for c in range(6, 0, -1):
  print(c)
print('FIM')

6
5
4
3
2
1
FIM


In [ ]:
n = int(input('Digite um número: '))
#f = int(input)

Digite um número: 56


In [ ]:
s = 0
for c in range(0,4):
  n = int(input('Digite um valor: '))
  s = s + n
print('O somatório de todos os valores foi {}'.format(s))

Digite um valor: 43
Digite um valor: 3
Digite um valor: 7
Digite um valor: 2
O somatório de todos os valores foi 55


# Desafio 046

Faça um programa que mostre na tela **contagem regressiva** para o estouro de fogos de artifício, indo de **10 até 0**, com uma pausa de **1 segundo** entre eles.


In [ ]:
import time

for c in range(10,-1, -1):
  print(c)
  time.sleep(1)

10
9
8
7
6
5
4
3
2
1
0


In [1]:
# Answer by Gustavo Guanabará
from time import sleep
for cont in range(10, -1, -1): # (0,11,1)
  print(cont)
  sleep(0.5)
print('BUM! BUM! POOOW!')

10
9
8
7
6
5
4
3
2
1
0
BUM! BUM! POOOW!


# Desafio 047

Crie um programa que mostre na tela **todos os números pares** que estão no intervalo de **1** e **50**

In [ ]:
for c in range(0,50):
  if c%2==0:
    print(c)

0
2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48


In [5]:
#Solved by Gustavo Guanabará
for n in range(1, 51, 2):
  # print('.', end='')
  # if n % 2 == 0: #print(n, end=' ')
    #print(n, end=' ')
  print(n, end=' ') # tempo do processor métode do tempo! 
print('Acabou')

1 3 5 7 9 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 Acabou


# Desafio 048

Faça um programa que calcule a **soma** entre todos os **número ímpares** que são **múltiplos de três** e que se encontram no intervalo de **1** até **500**


In [ ]:
sum = 0
for c in range(1, 500):
  if c %3 == 0 and c %2 ==1:
    sum += c
    print(c)
print(sum)
# print('termo médio vezes 2, ou os dois do meio se guardar num vetor, -1 ')



3
9
15
21
27
33
39
45
51
57
63
69
75
81
87
93
99
105
111
117
123
129
135
141
147
153
159
165
171
177
183
189
195
201
207
213
219
225
231
237
243
249
255
261
267
273
279
285
291
297
303
309
315
321
327
333
339
345
351
357
363
369
375
381
387
393
399
405
411
417
423
429
435
441
447
453
459
465
471
477
483
489
495
20667


In [12]:
# Solution by Gustavo Guanabará

soma = cont = 0
for c in range(1, 501, 2): # números ímpares!
  if c % 3 == 0:
    cont = cont + 1
    soma = soma + c
   # print(c, end=' ')
print('A soma de todos os {} valores solicitados é {}'.format(cont, soma))

A soma de todos os 83 valores solicitados é 20667


# Desafio 049

Refaça o **DESAFIO 009**, mostrando a **tabuada** de um número que o usuário escolher, só que agora utilizando um **laço for**.


In [ ]:
tabuada = int(input('Digite o número da tabuada que deseja: '))
print('\n Tabuada do {}'.format(tabuada))

print('\n')
print('-' * 39)

for c in range(0, 11):
  print('{} * {} = {}'.format(tabuada, c, tabuada*c))


Digite o número da tabuada que deseja: 5

 Tabuada do 5


---------------------------------------
5 * 0 = 0
5 * 1 = 5
5 * 2 = 10
5 * 3 = 15
5 * 4 = 20
5 * 5 = 25
5 * 6 = 30
5 * 7 = 35
5 * 8 = 40
5 * 9 = 45
5 * 10 = 50


In [14]:
# Answer by Gustavo Guanabará

num = int(input('Digite um número para ver sua tabuada: '))
for c in range(1,11):
  print('{} * {:2} = {}'.format(num, c, num*c))

Digite um número para ver sua tabuada: 7
7 *  1 = 7
7 *  2 = 14
7 *  3 = 21
7 *  4 = 28
7 *  5 = 35
7 *  6 = 42
7 *  7 = 49
7 *  8 = 56
7 *  9 = 63
7 * 10 = 70


# Desafio 050

Desenvolva um programa que leia **seis números inteiros** e mostre a soma apenas daqueles que forem **pares**. Se o valor digitado for **ímpar**, desconsidere-o.

In [ ]:
sum = 0
num={}
for c in range(1, 7):
  num[c] = int(input('Digit the number that you pretend: '))
  if num[c]%2 == 0:
    sum += num[c]
print(sum)


Digit the number that you pretend: 22
Digit the number that you pretend: 1
Digit the number that you pretend: 3
Digit the number that you pretend: 6
Digit the number that you pretend: 8
Digit the number that you pretend: 10
46


In [19]:
# Answered by Gustavo Guanabará

soma = cont = 0

for c in range(1,7):
  num = int(input('Digite o {} valor: '.format(c)))
  if num % 2 == 0:
    soma = soma + num
    cont += 1
print('Você informou {} números e a soma foi {}'.format(cont, soma))

Digite o 1 valor: 6
Digite o 2 valor: 9
Digite o 3 valor: 11
Digite o 4 valor: 12
Digite o 5 valor: 13
Digite o 6 valor: 8
Você informou 3 números e a soma foi 26


# Desafio 051

Desenvolve um programa que leia o **primeiro termo** e a **razão** de uma **PA**. No final, mostre os 10 primeiros termos dessa progressão.

In [38]:
# PA é progressão aritmética, então vem da soma. razão de 4 e primeiro termo seja 2, P1 será 6, P2 -> 10 etc

# base cases
prog = {}
prog[0] = int(input("Digit firST term: "))
ratio = int(input("Please, digit the ratio of formula: "))

print('0º -> {}'.format(prog[0]))
for c in range(1,11):
  prog[c] = prog[c-1] + ratio
  print('{}º -> {}'.format(c, prog[c]))

Digit firST term: 7
Please, digit the ratio of formula: 8
0º -> 7
1º -> 15
2º -> 23
3º -> 31
4º -> 39
5º -> 47
6º -> 55
7º -> 63
8º -> 71
9º -> 79
10º -> 87


In [39]:
# Solução Híbrida, feita à posteriori

primeiro = int(input('Primeiro termo: '))
razao = int(input('Razão: '))
#décimo = primeiro + (10-1)* razao
enésimo = int(input('How until term you want?: '))
for c in range(primeiro, primeiro + enésimo*razao, razao): # se o step for a razão fica muito acessível ...
  print('{} '.format(c), end=' ➙ ')
print('ACABOU')

Primeiro termo: 5
Razão: 6
How until term you want?: 3
5  ➙ 11  ➙ 17  ➙ ACABOU


In [40]:
# Solution by Gustavo Guanabara

# tiramos o elefante branco da sala

primeiro = int(input('Primeiro termo: '))
razao = int(input('Razão: '))
décimo = primeiro + (10-1)* razao
for c in range(primeiro, décimo + razao, razao): # se o step for a razão fica muito acessível ...
  print('{} '.format(c), end=' ➙ ')
print('ACABOU')

Primeiro termo: 25
Razão: 5
25  ➙ 30  ➙ 35  ➙ 40  ➙ 45  ➙ 50  ➙ 55  ➙ 60  ➙ 65  ➙ 70  ➙ ACABOU


# Desafio 052

 Faça um programa que leia um **número inteiro** e diga se ele é ou não um **número primo**.
 


In [26]:
import math
from math import sqrt
num = int(input('Digit a number: '))
primo = True 
if num == 0 or num == 1:
  primo = False
for c in range(2, int(math.sqrt(num))):
  primo = False
  break
print('É primo: {}'.format(primo))


Digit a number: 1
É primo: False


In [48]:
# Answered by Gustavo Guanabará

num = int(input('Digite um número: '))
tot = 0
for c in range(1, num +1):
  if num % c == 0:
    print('\033[33m', end='')
    tot += 1
  else:
    print('\033[31m', end='')
  print('{} '.format(c), end='')
print('\n\033[mO número {} foi divisível {} vezes'.format(num, tot))
if tot == 2:
  print('E por isso ele É PRIMO!')
else:
  print('É por isso ele NÃO É PRIMO!')
 # print('{} '.format(c), end='')

Digite um número: 67
1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 
O número 67 foi divisível 2 vezes
E por isso ele É PRIMO!


# Desafio 053

Crie um programa que leia uma **frase** qualquer e diga se ela é um **palídromo**,
desconsiderando os espaços.

Ex:

APOS A SOPA

A SACADA DA CASA

A TORRE DA DERROTA

 O LOBO AMA O BOLO

ANOTARAM A DATA DA MARATONA

In [9]:
import sys

phrase = str(input('Digit the phrase: '))
print(len(phrase))
phrase = phrase.strip()
phrase = phrase.replace(" ", "")
mid = int(len(phrase)/2)
print(phrase)

for c in range(0, mid):
  if phrase[c] != phrase[len(phrase)-c-1]:
    #print('Not palindrome')
    sys.exit('Not palindrome')
print('It\'s PALINDROME')

Digit the phrase: O LOBO AMA O BOLO
17
OLOBOAMAOBOLO
It's PALINDROME


In [2]:
# Answered by Gustavo Guanabara

# Tradicional version

frase = str(input('Digite uma frase: ')).strip().upper()
palavras = frase.split()
junto = ''.join(palavras) #'*'.join(palavras)
#print(junto)#palavras)
inverso = ''
inverso = junto[::-1] # fatiamento ou macete do fatiamento do python
''' for letra in range(len(junto)-1, -1, -1): # vai para o último index como começa em zero, temos de tirar 1 ..., depois vai até ao início que é 0 mas como é exclusivo .. temos de ir 1a além dessa: daí o -1,  como vai no sentido inverso do habitual then this!
   #print(junto[letra])
   inverso += junto[letra] '''
print('O inverso de {} é {}'.format(junto, inverso))
if inverso == junto:
  print('Temos um palíndromo!')
else:
  print('A frase digitada não é um palíndromo!')
# print('Você digitou a frase {}'.format(#junto))

Digite uma frase: Hello Mocho!
O inverso de HELLOMOCHO! é !OHCOMOLLEH
A frase digitada não é um palíndromo!


In [25]:
type(int(len(phrase)/2))

int

# Desafio 054

Crie um programa que leia o **ano de nascimento** de **sete pessoas**. No final, mostre quantas pessoas ainda não atingiram a maioridade e quantas já são maiores.

In [25]:
def clean_date_of_birth(self):
    date = self.cleaned_data['date_of_birth']
    if date > datetime.date.today():  # 🖘 raise error if greater than
        raise forms.ValidationError("The date cannot be in the future!")
    return date

In [3]:
from datetime import date
adults = 0
tries = childs = 0
ano = {}
actualYear = date.today().year
print(actualYear)
for c in range(0,7):
  ano[c] = int(input('Digit the bornYear: '))
  while ano[c] > actualYear:
    ano[c] = int(input('Digit the bornYear: '))
    tries += 1
  # while (date.today() < ano[c]):
  #   ano[C] = int(input('Digit the bornYear: '))
  print(ano[c])
  if actualYear - ano[c] >= 21:
    adults += 1
  else:#if actualYear - ano[c]< 0:  # garantir que não tem datas futuras... 
    childs += 1
print('Number of Babys: {} \n Number of Adults: {}'.format(childs, adults))


2022
Digit the bornYear: 2024
Digit the bornYear: 2022
2022
Digit the bornYear: 2021
2021
Digit the bornYear: 2023
Digit the bornYear: 2021
2021
Digit the bornYear: 2000
2000
Digit the bornYear: 2222
Digit the bornYear: 221
221
Digit the bornYear: 2012
2012
Digit the bornYear: 2013
2013
Number of Babys: 5 
 Number of Adults: 2


In [22]:
#install python 3.9
!sudo apt-get update -y
!sudo apt-get install python3.9

#changes alternatives
#!python --version


Get:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease [1,581 B]
Get:2 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ InRelease [3,626 B]
Get:3 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic InRelease [15.9 kB]
Get:4 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]
Ign:5 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease
Hit:6 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release
Hit:7 http://archive.ubuntu.com/ubuntu bionic InRelease
Get:8 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Packages [817 kB]
Get:9 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
Get:10 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ Packages [85.6 kB]
Hit:11 http://ppa.launchpad.net/cran/libgit2/ubuntu bionic InRelease
Hit:13 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu bioni

In [25]:
# changes alternatives
!sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.7 1
!sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.9 2

update-alternatives: using /usr/bin/python3.9 to provide /usr/bin/python3 (python3) in auto mode


In [27]:
from datetime import date
adults = 0
childs = 0
ano = {}
actualYear = date.today().year
print(actualYear)
for c in range(0,7):
  while actualYear < (bornYear := int(input('Digit the bornYear: '))): 
  # while (date.today() < ano[c]):
  #   ano[C] = int(input('Digit the bornYear: '))
  print(ano[c])
  if actualYear - ano[c] >= 21:
    adults += 1
  else:#if actualYear - ano[c]< 0:  # garantir que não tem datas futuras... 
    childs += 1
print('Number of Babys: {} \n Number of Adults: {}'.format(childs, adults))

SyntaxError: ignored

In [26]:
!python --version

Python 3.9.13


In [9]:
# Answered by Gustavo Guanabará

from datetime import date
atual = date.today().year
totmaior = 0
totmenor = 0
for pess in range(1,8):
  nasc = int(input('Em que ano a {}ª pessoa nasceu? '.format(pess)))
  idade = atual - nasc
  if idade >= 21:
    #print('Essa pessoa é maior')
    totmaior += 1
  else:
    #print('Essa pessoa é menor')
    totmenor += 1
print('Ao todo tivemos {} pessoas maiores de idade'.format(totmaior))
print('Ao todo tivemos {} pessoas menores de idade'.format(totmenor))

Em que ano a 1ª pessoa nasceu? 3
Em que ano a 2ª pessoa nasceu? 5
Em que ano a 3ª pessoa nasceu? 6
Em que ano a 4ª pessoa nasceu? 7
Em que ano a 5ª pessoa nasceu? 6
Em que ano a 6ª pessoa nasceu? 5674
Em que ano a 7ª pessoa nasceu? 4365
Ao todo tivemos 5 pessoas maiores de idade
Ao todo tivemos 2 pessoas menores de idade


# Desafio 055

Faça um programa que leia o **peso** de **cinco pessoas**. No final, mostre qual foi o **maior** e o **menor** peso lidos.

In [29]:
weight = {}
min = 1000
max = 0 
for c in range(0, 5): 
  weight[c] = int(input('Digit the person weight: '))
  if weight[c] < min:
    min = weight[c]
  if max < weight[c]: # nada a ver , mas se todos iguais não atualiza ...
    max = weight[c]
print('Os pesos máx. = {} e min. = {}'.format(max, min))

Digit the person weight: 120
Digit the person weight: 56
Digit the person weight: 78
Digit the person weight: 75
Digit the person weight: 90
Os pesos máx. = 56 e min. = 120


In [35]:
import numpy as np

In [43]:
list1 = [67, 53, 88, 39, 102]
# weight = {67, 53, 88, 39, 102}
weight = np.array(list1)
min = max = weight[0]
for c in range(0, len(weight)):
  if min > weight[c]:
    min = weight[c]
  elif weight[c] > max:
    max = weight[c]
print('The weights maximum is {} and the minimum is {}'.format(max, min)) 


The weights maximum is 102 and minimum is 39


In [44]:
weight[0]

67

In [12]:
# Solution by Gustavo Guanabara

# Análise de dados, área de guarde importância no momentoi (totaliza, outra count .. ou diz o maior e menor [extremos] de uma lista)
maior = 0
menor = 0
for p in range(1,6):
  peso = float(input('Peso da {}ª pessoa: '.format(p)))
  if p == 1:
    maior = peso
    menor = peso
  else:
    if peso > maior:
      maior = peso
    if peso < menor:
      menor = peso
print('O maior peso lido foi de {}Kg'.format(maior))
print('O menor peso lido foi de {}Kg'.format(menor))

Peso da 1ª pessoa: 45
Peso da 2ª pessoa: 75
Peso da 3ª pessoa: 98
Peso da 4ª pessoa: 34
Peso da 5ª pessoa: 86
O maior peso lido foi de 98.0Kg
O menor peso lido foi de 34.0Kg


In [11]:
p

5

# Desafio 056

Desenvolva um programa que leia o **nome**, **idade** e **sexo** de **4 pessoas**. No final do programa, mostre:



*   A **média de idade** do grupo.
*   Qual é o nome do homem **mais velho**.
*   Quantas mulheres têm **menos de 20 anos**.



In [18]:
# names = {}
# ages = {}
# #sex = []
# sex = {}
# maXman = sum = count = max = 0
# maXmanI = -1 # saber o nome do homem mais velho..., implica saber o index do man
# S = 4
# for c in range(0, S):
#   names[c] = str(input(' Digit person name: '))
#   ages[c] = int(input('Digit person age: '))
#   sex[c] = str(input('Please, digit your sex: F or M?: ')).strip().upper()[0]
#   sum += ages[c]
#   if sex == 'F' and ages[c]<20:  # sex.upper().strip() == 'F' and ages[c]<20:
#   #if sex.strip() == 'F' and ages[c]<20:
#     count += 1
#   if sex == 'M': # .upper()
#     print(ages[c])
#     if (max<ages[c]):
#       max = ages[c]
#       maXmanI = c
# mean = sum / S
# print(c)
# print(maXmanI)
# print(count)
# print('A média de idades é: {}, o homem mais velho é: {} e\n existem {} mulheres com menos de 20 anos!'.format(mean,names[maXmanI], count))     



 Digit person name: p
Digit person age: 09
Please, digit your sex: F or M?: f
 Digit person name: c
Digit person age: 99
Please, digit your sex: F or M?: m
 Digit person name: cat
Digit person age: 17
Please, digit your sex: F or M?: f
 Digit person name: xavi
Digit person age: 33
Please, digit your sex: F or M?: m
3
-1
0


KeyError: ignored

In [47]:
type(sex)

dict

In [30]:
# Answer by Gustavo Guanabara
somaidade = 0
meidaidade = 0
maioridadehomem = totmulher20 = 0
nomevelho = ''
for p in range(1, 5):
  print('------ {}ª PESSOA ------'.format(p))
  nome = str(input('Nome: ')).strip()
  idade = int(input('Idade: '))
  sexo = str(input('Sexo [M/F]: ')).strip()
  somaidade += idade
  if p == 1 and sexo in 'Mm':
    maioridadehomem = idade
    nomevelho = nome 
  if sexo in 'Mm' and idade > maioridadehomem:
    maioridadehomem = idade
    nomevelho = nome
  if sexo in 'Ff' and idade < 20: #interessante, cadeia de string para definir.., evita a utilização de upper etc
    totmulher20 += 1
mediaidade = somaidade / 4
print('A média de idade do grupo é de {}.'.format(mediaidade))
print('O homem mais velho tem {} anos e se chama {}.'.format(maioridadehomem, nomevelho))
print('Ao todo são {} mulheres com menos de 20 anos'.format(totmulher20))

------ 1ª PESSOA ------
Nome: Ju
Idade: 12
Sexo [M/F]: F
------ 2ª PESSOA ------
Nome: Gasd
Idade: 30
Sexo [M/F]: M
------ 3ª PESSOA ------
Nome: ffg
Idade: 19
Sexo [M/F]: F
------ 4ª PESSOA ------
Nome: hugo
Idade: 21
Sexo [M/F]: F
A média de idade do grupo é de 20.5.
O homem mais velho tem 30 abnos e se chama Gasd.
Ao todo são 2 mulheres com menos de 20 anos


In [ ]:
# For : estrutura de repetição com variável de controlo <- qdo tem limite
# While : estrutura de repetição (não existe um padrão, está muito irregular!)
# não existem mais rápido entre os dois

In [1]:
# Aula 14

for c in range(1,10): # Quando não sei o limite não dá para usar o FOR
  print(c)
print('Fim')



1
2
3
4
5
6
7
8
9
Fim


In [2]:
c = 1
while c < 10: 
  print(c)
  c = c + 1
print('Fim')

1
2
3
4
5
6
7
8
9
Fim


In [3]:
n = 1
while n != 0:  # FLAG ou condição de parada
  n = int(input('Digite um valor: '))
print('Fim')

Digite um valor: 5
Digite um valor: 8
Digite um valor: 0
Fim


In [4]:
r = 'S'
while r == 'S': # faz laços indetermininadamente
  n = int(input('Digite um valor: '))
  r = str(input('Quer continuar? [S/N] ')).upper()  
print('Fim')

Digite um valor: 45
Quer continuar? [S/N] S
Digite um valor: 323
Quer continuar? [S/N] 7
Fim


In [ ]:
n = 1
par = impar = 0
while n != 0:
  n = int(input('Digite um valor: '))
  if n != 0: # gambiarra
    if n % 2 == 0:
      par += 1 
    else:
      impar += 1
print('Voçê digitou {} números pares e {} números ímpares!'.format(par, impar))

# Desafio 57

Faça um programa que leia o **sexo** de uma pessoa, mas só aceite os valores 'M' ou 'F'. 
Caso esteja errado, peça a digitação novamente até ter um valor correto.

In [13]:
gender = str(input('Say your sex: '))  
while gender not in 'Ff' and gender not in 'mM':
  gender = str(input('Say your sex: '))  
print('Sex {} registered with sucess.'.format(gender))

Say your sex: h
Say your sex: d
Say your sex: f
Sex f registered with sucess.


In [23]:
# answer by Gustavo Guanabará
sexo = str(input('Informe seu sexo: [M/F] ')).strip().upper()[0] # nenhuma validação até então
#idae = int(input('Informe sua idade: '))
while( sexo not in 'MF'): #'MmFf'):
  sexo = str(input('Dados inválidos. Por favor, informe seu sexo: ')).strip().upper()[0]
print('Sexo {} registado com sucesso'.format({sexo}))

Informe seu sexo: [M/F] f
Sexo {'F'} registado com sucesso


**Desafio 58**

Melhore o jogo do **Desafio 028** onde o computador vai "pensar" em um **número entre 0 e 10**. Só que agora o jogador vai tentar adivinhar até acertar, mostrando no final quantos palpites foram necessários para vencer.

In [7]:
import random
count = 1
num = random.randint(0, 11)
unum = int(input('Digite um número entre 0 e 10: '))
while num != unum :
  unum = int(input('Digite um número entre 0 e 10: '))
  count += 1
print('The player guess the num in {} tries'.format(count))

Digite um número entre 0 e 10: 6
Digite um número entre 0 e 10: 9
Digite um número entre 0 e 10: 1
Digite um número entre 0 e 10: 0
Digite um número entre 0 e 10: 3
Digite um número entre 0 e 10: 2
The player guess the num in 6 tries


In [24]:
# Answer by Gustavo Guanabará
from random import randint
computador = randint(0,10)
print('Sou seu computador... Acabei de pensar em um número entre 0 e 10.')
print('Será que voçê consegue adivinhar qual foi? ')
acertou = False
palpites = 0
while not acertou:
  jogador = int(input('Qual é seu palpite? '))
  palpites += 1
  if jogador == computador:
    acertou = True
  else:
    if jogador < computador: # input('Qual é seu palpite')
      print('Mais... Tente mais uma vez.')
    elif jogador > computador:
      print('Menos... Tente mais uma vez.')
print('Acertou com {} tentativas. Parabéns!'.format(palpites))


Sou seu computador... Acabei de pensar em um número entre 0 e 10.
Será que voçê consegue adivinhar qual foi? 
Qual é seu palpite? 321
Menos... Tente mais uma vez.
Qual é seu palpite? 222
Menos... Tente mais uma vez.
Qual é seu palpite? 10
Menos... Tente mais uma vez.
Qual é seu palpite? 9
Menos... Tente mais uma vez.
Qual é seu palpite? 4
Mais... Tente mais uma vez.
Qual é seu palpite? 5
Acertou com 6 tentativas. Parabéns!


# Desafio 059

Crie um programa que leia **dois valores** e mostre um **menu** na tela:



[1]    Somar

[2]    Multiplicar

[3]    Maior

[4]    Novos números

[5]    Sair do programa


Seu programa deverá realizar a operação solicitada em cada caso.


In [2]:
def readNums():
  a = int(input('Digit a number: '))
  b = int(input('Digit other number: '))
  return a, b

In [18]:
# def ciclo(option):
#     if option == 1:
#       print('Sum: {} + {} = {}'.format(num1, num2, num1+num2))
#     elif option == 2:
#       print('Multiplication: {} * {} = {}'.format(num1, num2, num1*num2))
#     elif option == 3:
#       print('Greater is {}'.format(max(num1, num2)))
#     elif option == 4:
#       num1, num2 = readNums()
#     elif option == 5:
#       print('The End!')
#       break

SyntaxError: ignored

In [3]:
num1, num2 = readNums()
while True:
  print('''\nWelcome! 
          \nChoose a option:\n
            [1] Sum
            [2] Multiplication
            [3] Greater
            [4] New numbers
            [5] Exit \n\n''')
  option = int(input('\nSay a option: '))
            
  while option > 6 or option < 0:
    option = int(input('Invalid option ...\nSay a option: '))
  print(f'\n[{option}]')
  if option == 1:
    print('Sum: {} + {} = {}'.format(num1, num2, num1+num2)) 
  elif option == 2:
    print('Multiplication: {} * {} = {}'.format(num1, num2, num1*num2)) 
  elif option == 3:
    # max(num1, num2))
    print('Greater is {}'.format(max(num1, num2))) 
  elif option == 4: 
    num1, num2 = readNums() 
  elif option == 5: 
    print('\nThe End!') 
    break
  


Digit a number: 5
Digit other number: 3

Welcome! 
          
Choose a option:

            [1] Sum
            [2] Multiplication
            [3] Greater
            [4] New numbers
            [5] Exit 



Say a option: 2

[2]
Multiplication: 5 * 3 = 15

Welcome! 
          
Choose a option:

            [1] Sum
            [2] Multiplication
            [3] Greater
            [4] New numbers
            [5] Exit 



Say a option: 4

[4]
Digit a number: 4
Digit other number: 7

Welcome! 
          
Choose a option:

            [1] Sum
            [2] Multiplication
            [3] Greater
            [4] New numbers
            [5] Exit 



Say a option: 1

[1]
Sum: 4 + 7 = 11

Welcome! 
          
Choose a option:

            [1] Sum
            [2] Multiplication
            [3] Greater
            [4] New numbers
            [5] Exit 



Say a option: 3

[3]
Greater is 7

Welcome! 
          
Choose a option:

            [1] Sum
            [2] Multiplication
            [3] G

In [43]:
del max

In [39]:
import math

In [44]:
nums = [8,5,3,1,6]
big = max(nums);
print(big)

8


In [17]:
# num1, num2 = readNums()
# # option = 1
# # while (option > 0):
# #   option = int(input('Say a option: '))
# option = 4
# while (option > 5): # necesaário se quiser garantir que quando vai fora do scope de 1 a cinco.., faz repetir a opção
#   option = int(input('Say a option: '))
#   if option > 0 and option < 6: # ao invés do while
#     option = int(input('Say a option: '))
#     if option == 1:
#       print('Sum: {} + {} = {}'.format(num1, num2, num1+num2))
#     elif option == 2:
#       print('Multiplication: {} * {} = {}'.format(num1, num2, num1*num2))
#     elif option == 3:
#       print('Greater is {}'.format(max(num1, num2)))
#     elif option == 4:
#       num1, num2 = readNums()
#     elif option == 5:
#       print('The End!')
#       break
#     # else:
#     #   print('The End!')
#     #   break

Digit a number: 56
Digit other number: 12
Say a option: 2
Multiplication: 56 * 12 = 672
Say a option: 1
Sum: 56 + 12 = 68
Say a option: 3
Greater is 56
Say a option: 4
Digit a number: 43
Digit other number: 78
Say a option: 1
Sum: 43 + 78 = 121
Say a option: 7


In [48]:
# Answer by Gustavo Guanabará

n1 = int(input('Primeiro valor: '))
n2 = int(input('Segundo valor: '))
opção = 0
while opção != 5:
  print('''  [ 1 ] somar
  [ 2 ] multiplicar
  [ 3 ] maior
  [ 4 ] novos números
  [ 5 ] sair do programa''')
  opção = int(input('>>>>> Qual é a sua opção? '))
  if opção == 1:
    soma = n1 + n2
    print('A soma entre {} + {} é {}'.format(n1, n2, soma))
  elif opção == 2:
    produto = n1 * n2
    print('O resultado de {} * {} é {}'.format(n1, n2, produto))
  elif opção == 3:
    if n1 > n2:
      maior = n1
    else:
      maior = n2
    print('Entre {} e {} o maior valor é {}'.format(n1, n2, maior))
  elif opção == 4:
    print('Informe os números novamente: ')
    n1 = int(input('Primeiro valor: '))
    n2 = int(input('Segundo valor: '))
  elif opção == 5:
    print('Finalizando...')
  else: 
    print('Opção inválida. Tente novamente')
  print('=-=' * 10)
print('Fim do programa! Volte sempre!')

Primeiro valor: 2
Segundo valor: 5
   [ 1 ] somar
  [ 2 ] multiplicar
  [ 3 ] maior
  [ 4 ] novos números
  [ 5 ] sair do programa
>>>>> Qual é a sua opção? 1
A soma entre 2 + 5 é 7
=-==-==-==-==-==-==-==-==-==-=
   [ 1 ] somar
  [ 2 ] multiplicar
  [ 3 ] maior
  [ 4 ] novos números
  [ 5 ] sair do programa
>>>>> Qual é a sua opção? 2
O resultado de 2 * 5 é 10
=-==-==-==-==-==-==-==-==-==-=
   [ 1 ] somar
  [ 2 ] multiplicar
  [ 3 ] maior
  [ 4 ] novos números
  [ 5 ] sair do programa
>>>>> Qual é a sua opção? 3
Entre 2 e 5 o maior valor é 5
=-==-==-==-==-==-==-==-==-==-=
   [ 1 ] somar
  [ 2 ] multiplicar
  [ 3 ] maior
  [ 4 ] novos números
  [ 5 ] sair do programa
>>>>> Qual é a sua opção? 4
Informe os números novamente: 
Primeiro valor: 6
Segundo valor: 1
=-==-==-==-==-==-==-==-==-==-=
   [ 1 ] somar
  [ 2 ] multiplicar
  [ 3 ] maior
  [ 4 ] novos números
  [ 5 ] sair do programa
>>>>> Qual é a sua opção? 1
A soma entre 6 + 1 é 7
=-==-==-==-==-==-==-==-==-==-=
   [ 1 ] somar
  [ 2 ]

# Desafio 60

Faça um programa que leia um **número** qualquer e mostre o seu **fatorial**.

Ex:
5! = 5 * 4 * 3 * 2 * 1 = 120 





In [59]:
def fatorial(n):
  if n == 1:
    print('{} \n='.format(n))
    return 1
  else:
    print('{} \n*'.format(n , end = ' '))
    return (n * fatorial(n-1))

In [77]:
fat = int(input('Say fatorial number that you wish: '))
print('{} = {}!'.format(fatorial(fat), fat))

Say fatorial number that you wish: 10
10 
*
9 
*
8 
*
7 
*
6 
*
5 
*
4 
*
3 
*
2 
*
1 
=
3628800 = 10!


In [78]:
def fatorial2 (fat): 
  result = 1 # In multiplication, identity element is 1, division to
  while fat > 0:
    print('{} \n*'.format(fat , end = ' '))
    result *= fat
    fat -= 1
  return result



In [79]:
print('{} = {}!'.format(fatorial2(fat), fat))

10 
*
9 
*
8 
*
7 
*
6 
*
5 
*
4 
*
3 
*
2 
*
1 
*
3628800 = 10!


In [80]:
def fatorial3 (fat): 
  result = 1 # In multiplication, identity element is 1, division to
  for n in range(1,fat+1):
    print('{} \n*'.format(n , end = ' '))
    result *= n
  return result

In [81]:
def fatorial4 (fat): 
  result = 1 # In multiplication, identity element is 1, division to
  for n in range(fat,0, -1):
    print('{} \n*'.format(n , end = ' '))
    result *= n
  return result

In [82]:
print('{} = {}!'.format(fatorial3(fat), fat))

1 
*
2 
*
3 
*
4 
*
5 
*
6 
*
7 
*
8 
*
9 
*
10 
*
3628800 = 10!


In [83]:
print('{} = {}!'.format(fatorial4(fat), fat))

10 
*
9 
*
8 
*
7 
*
6 
*
5 
*
4 
*
3 
*
2 
*
1 
*
3628800 = 10!


In [50]:
# Answer by Gustavo Guanabará
from math import factorial
n = int(input('Digite um número para calcular seu Fatorial: '))
f = factorial(n)
print('O fatorial de {} é {}.'.format(n, f))


# em 4 linhas

Digite um número para calcular seu Fatorial: 5
O fatorial de 5 é 120.


In [56]:
from math import factorial
n = int(input('Digite um número para calcular seu Fatorial: '))
c = n
f = 1
print('Calculando {}! = '.format(n), end='')
while c > 0: # serve para qdo sei e não sei limite, for only quando sei
# f = factorial(n)
  #print('O fatorial de {} é {}.'.format(n, f))
  print('{}'.format(c), end='')
  print(' x ' if c > 1 else ' = ', end='')
  f *= c
  c -= 1
print(f) #print('{}'.format(f))
# print('O fatorial de {} é {}.'.formal)
# em 4 linhas, o GUanabará faz estes exercícios no âmbito de sala de aula!!

Digite um número para calcular seu Fatorial: 8
Calculando 8! = 8 x 7 x 6 x 5 x 4 x 3 x 2 x 1 = 40320


# Desafio 061

Refaça o **Desafio 051**, lendo o **primeiro termo** e a **razão** de uma **PA**, mostrando os **10 primeiros termos** da progressão usando a estrutura **while**

In [92]:
pav = basis = float(input("Digit first term: \n"))

ratio = float(input("Say ratio that you want: \n"))

count = 0

print('The first 10 term of PA is: \n')
while count < 10:
  pav += ratio
  count += 1
  print('{:.3f}'.format(pav))

Digit first term: 
1.51
Say ration that you want: 
0.1
The first 10 term of PA is: 

1.610
1.710
1.810
1.910
2.010
2.110
2.210
2.310
2.410
2.510


In [59]:
# Solution of Prof. Gustavo Guanabará

print(' Gerador de PA')
print('-= * 10')
primeiro = int(input('Primeiro termo: '))
razão = int(input('Razão da PA: '))
termo = primeiro
cont = 1
while cont <= 10:
  print('{} ➙ '.format(termo), end='')
  termo += razão
  cont+=1
print('FIM') # termo/fundamento para calcular o décimo termo, que definia o limite exigido no for ...na aula 51, aqui utilizou lógico, contadores para termo e contador de vezes!

 Gerador de PA
-= * 10
Primeiro termo: 5
Razão da PA: 3
5 ➙ 8 ➙ 11 ➙ 14 ➙ 17 ➙ 20 ➙ 23 ➙ 26 ➙ 29 ➙ 32 ➙ FIM


#Desafio 062
Melhore o **Desafio 061**, perguntando para o usuário se ele equer mostrar mais alguns termos. O programa encerra quando ele disser que quer mostrar **0 termos**.

In [10]:
def aterms(base, countElem,  ratio):
  # end = base + countElem * ratio
  while(countElem > 0):
    print('{:.3f}'.format(base))
    base += ratio
    countElem -= 1
  return base # end

In [11]:
basis = float(input("Digit first term: \n"))

ratio = float(input("Say ratio that you want: \n"))

countElem = 10


print('The first 10 term of PA is: \n')
while countElem > 0:
  basis = aterms(basis, countElem, ratio)
  countElem = int(input('How much more you want to see:'))
# while count < MACRO:
#   pav += ratio
#   count += 1
#   print('{:.3f}'.format(pav))



Digit first term: 
34
Say ratio that you want: 
2
The first 10 term of PA is: 

34.000
36.000
38.000
40.000
42.000
44.000
46.000
48.000
50.000
52.000
How much more you want to see:4
54.000
56.000
58.000
60.000
How much more you want to see:8
62.000
64.000
66.000
68.000
70.000
72.000
74.000
76.000
How much more you want to see:-1


In [ ]:
# def qtd+():
#   dif = int(input('How much more you want? '))
#   return dif

In [5]:
frase = "10234540234 the"

def get_firstWord(frase):
  fraseSplitted = frase.split()
  return fraseSplitted[0]

get_firstWord(frase)

'the'

In [8]:
# def aterms(base, countElem,  ratio):
#   end = base + countElem * ratio
#   while(countElem > 0):
#     print('{:.3f}'.format(base))
#     base += ratio
#     countElem -= 1
#   return  end

In [9]:
# basis = float(input("Digit first term: \n"))

# ratio = float(input("Say ratio that you want: \n"))

# countElem = 10


# print('The first 10 term of PA is: \n')
# while countElem > 0:
#   basis = aterms(basis, countElem, ratio)
#   countElem = int(input('How much more you want to see:'))
# # while count < MACRO:
# #   pav += ratio
# #   count += 1
# #   print('{:.3f}'.format(pav))

Digit first term: 
34
Say ratio that you want: 
2
The first 10 term of PA is: 

34.000
36.000
38.000
40.000
42.000
44.000
46.000
48.000
50.000
52.000
How much more you want to see:4
54.000
56.000
58.000
60.000
How much more you want to see:0


In [4]:
# Solution by Prof. Gustavo Guanabará

print(' Gerador de PA')
print('-= * 10')
primeiro = int(input('Primeiro termo: '))
razão = int(input('Razão da PA: '))
termo = primeiro
cont = 1
total = 0
mais = 10
while mais != 0:
  total = total + mais
  while cont <= total:
    print('{} ➙ '.format(termo), end='')
    termo += razão
    cont += 1
  print('PAUSA')
  mais = int(input('Quantos termos voçê quer mostrar a mais? '))
print('Progressão finalizada com {} termos mostrados.'.format(total))

 Gerador de PA
-= * 10
Primeiro termo: 4
Razão da PA: 3
4 ➙ 7 ➙ 10 ➙ 13 ➙ 16 ➙ 19 ➙ 22 ➙ 25 ➙ 28 ➙ 31 ➙ PAUSA
Quantos termos voçê quer mostrar a mais? 7
34 ➙ 37 ➙ 40 ➙ 43 ➙ 46 ➙ 49 ➙ 52 ➙ PAUSA
Quantos termos voçê quer mostrar a mais? 3
55 ➙ 58 ➙ 61 ➙ PAUSA
Quantos termos voçê quer mostrar a mais? 0
Progressão finalizada com 20 termos mostrados.


# Desafio 063

Escreva um programa que leia um **número n** inteiro qualquer e mostre na tela os **n** primeiros elementos de uma **Sequência de Fibonacci**

In [25]:
def Fibonacci(n): # Forma recursiva ... mal feita ... see depois de pensar mais um pouco se sobrar.., algum tempo!
  b0 = 0
  b1 = 1
  if Fibonacci(0):
    return b0 + b1 # return
  elif Fibonacci(1):
    return Fibonacci(0) + b1
  else:
   while (n > 1):
    print('{}+'.format(Fibonacci(n)))
    return Fibonacci(n-1)+Fibonacci(n-2)
 


In [26]:
n = int(input('How many Fibonacci terms you want? '))
print(Fibonacci(n))

How many Fibonacci terms you want? 2


RecursionError: ignored

In [ ]:
# Solution of Gustavo Guanabara

# 0 - 1 - 1 - 2 - 3 - 5 - 8 - 13 ...

In [6]:
print('-' * 30)
print('Sequência de Fibonacci')
print('-'*30)
n = int(input('Quantos termos voçê quer mostrar? '))
t1 = 0
t2 = 1
print('~' * 30)
print('{} ➙ {}'.format(t1, t2), end='')
cont = 3
while cont <= n:
  t3 = t1 + t2
  print(' ➙ {}'.format(t3), end='')
  t1 = t2
  t2 = t3
  cont += 1
print(' ➙ FIM')


------------------------------
Sequência de Fibonacci
------------------------------
Quantos termos voçê quer mostrar? 12
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
0 ➙ 1 ➙ 1 ➙ 2 ➙ 3 ➙ 5 ➙ 8 ➙ 13 ➙ 21 ➙ 34 ➙ 55 ➙ 89 ➙ FIM


# Desafio 064

Crie um programa que leia **vários números inteiros pelo teclado. O programa só vai parar quando o usuário digitar o valor **999**, que é a **condição de parada**. No final, mostre **quantos números** foram digitados e qual foi  a **soma** entre eles *(desconsiderando o flag)*.

In [28]:
counter = s = 0
n = float(input('Digit a number: '))
while (n != 999):
  counter += 1
  s += n
  n = float(input('Digit a number: '))
print('{} numbers was typescripted.\n Sum =  {}'.format(counter, s))


Digit a number: 34
Digit a number: 34
Digit a number: 32
Digit a number: -9
Digit a number: 999
4 numbers was typescripted.
 Sum =  91.0


In [15]:
# Solution by Gustavo Guanabara
núm = soma = cont = 0
núm = int(input('Digite um número [999 para parar]: '))
while núm != 999:
 # núm = int(input('Digite um número [999 para parar]: '))
  cont += 1
  soma += núm
  núm = int(input('Digite um número [999 para parar]: '))
print('Voçê digitou {} números e a soma entre eles foi {}.'.format(cont, soma))#(cont-1, soma-999))#'Acabou')

Digite um número [999 para parar]: 999
Voçê digitou 0 números e a soma entre eles foi 0.


# Desafio 065

Crie um programa que leia **vários números** inteiros pelo teclado. No final da execução, mostre a **média entre todos** os valores e qual foi **maior** e o **menor** valores lidos. O programa deve perguntar ao usuário se ele quer ou não **continuar** a digitar valores.

In [11]:
continua = 1 #True
avg = sum = count = 0
max = min = num = int(input('Digit number: '))
while(continua):
  count += 1
  sum += num
  continua = int(input('Do you want read values? \n[1] Yes \n[0] NO! \n[*] yeah! \n Getting your answer ... :  '))# bool(input('Do you want read values? '))
  # if continua in 'fF':
  #   continua = False
  if (continua == 0):
    break
  num = int(input('\n-------------------\nDigit a number, please! '))
  if max < num:
    max = num
  elif num < min:
    min = num
avg = sum/count
print('\nThe average is {}.\nThe max value is {}.\nThe min value is {}'.format(avg, max, min))


Digit number: 32
Do you want read values? 
[1] Yes 
[0] NO! 
[*] yeah! 
 Getting your answer ... :  0

The average is 32.0.
The max value is 32.
The min value is 32


In [7]:
# Solution by Gustavo Guanabara

resp = 'S'
soma = quant = média = maior = menor = 0
while resp in 'Ss': # se for outra coisa qq, ele então pará
  núm = int(input('Digite um número: '))
  soma += núm
  quant += 1
  if quant == 1:
    maior = menor = núm
  else:
    if núm > maior:
      maior = núm
    elif núm < menor:
      menor=núm
  resp = str(input('Quer continuar? [S/N] ')).upper().strip()[0]
média = soma / quant
print('Voçê digitou {} números e a média foi {}'.format(quant, média))
print('O maior valor foi {} e o menor foi {}'.format(maior, menor))

Digite um número: 78
Quer continuar? [S/N] s
Digite um número: 89
Quer continuar? [S/N] s
Digite um número: 12
Quer continuar? [S/N] n
Voçê digitou 3 números e a média foi 59.666666666666664
O maior valor foi 78 e o menor foi 78


# Aula 15

In [ ]:
''' laços de repetição:

for 
while 

comando break ... que sai do ciclo (joga para fora de uma estrutura de repetição)

In [4]:
# Parte prática

cont = 1
while cont <= 10: #True: # # Carregar stop or o comando break para terminar o ... programa
  print(cont, '->', end='')
  cont += 1
print('Acabou') 

1 ->2 ->3 ->4 ->5 ->6 ->7 ->8 ->9 ->10 ->11 ->12 ->13 ->14 ->15 ->16 ->17 ->18 ->19 ->20 ->21 ->22 ->23 ->24 ->25 ->26 ->27 ->28 ->29 ->30 ->31 ->32 ->33 ->34 ->35 ->36 ->37 ->38 ->39 ->40 ->41 ->42 ->43 ->44 ->45 ->46 ->47 ->48 ->49 ->50 ->51 ->52 ->53 ->54 ->55 ->56 ->57 ->58 ->59 ->60 ->61 ->62 ->63 ->64 ->65 ->66 ->67 ->68 ->69 ->70 ->71 ->72 ->73 ->74 ->75 ->76 ->77 ->78 ->79 ->80 ->81 ->82 ->83 ->84 ->85 ->86 ->87 ->88 ->89 ->90 ->91 ->92 ->93 ->94 ->95 ->96 ->97 ->98 ->99 ->100 ->101 ->102 ->103 ->104 ->105 ->106 ->107 ->108 ->109 ->110 ->111 ->112 ->113 ->114 ->115 ->116 ->117 ->118 ->119 ->120 ->121 ->122 ->123 ->124 ->125 ->126 ->127 ->128 ->129 ->130 ->131 ->132 ->133 ->134 ->135 ->136 ->137 ->138 ->139 ->140 ->141 ->142 ->143 ->144 ->145 ->146 ->147 ->148 ->149 ->150 ->151 ->152 ->153 ->154 ->155 ->156 ->157 ->158 ->159 ->160 ->161 ->162 ->163 ->164 ->165 ->166 ->167 ->168 ->169 ->170 ->171 ->172 ->173 ->174 ->175 ->176 ->177 ->178 ->179 ->180 ->181 ->182 ->183 ->184 ->185 

ERROR:root:Internal Python error in the inspect module.
Below is the traceback from this internal error.



Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py", line 2882, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-4-45727f602465>", line 5, in <module>
    print(cont, '->', end='')
  File "/usr/local/lib/python3.7/dist-packages/ipykernel/iostream.py", line 400, in write
    self.pub_thread.schedule(lambda : self._buffer.write(string))
  File "/usr/local/lib/python3.7/dist-packages/ipykernel/iostream.py", line 203, in schedule
    self._event_pipe.send(b'')
  File "/usr/local/lib/python3.7/dist-packages/zmq/sugar/socket.py", line 618, in send
    return super().send(data, flags=flags, copy=copy, track=track)
  File "zmq/backend/cython/socket.pyx", line 740, in zmq.backend.cython.socket.Socket.send
  File "zmq/backend/cython/socket.pyx", line 787, in zmq.backend.cython.socket.Socket.send
  File "zmq/backend/cython/socket.pyx", line 244, in zmq.backend.cython.socket._send_copy
  File "z

KeyboardInterrupt: ignored

In [8]:
n = cont = s = 0 # inicializa variáveis no Python não se declaram (somente)...
while True: #n != 999 #cont < 5:3: # com limites (Qto nums são:!)
#while n != 999: Por flag ponto de parada! (fator condicional) 
  n = int(input('Digite um número: '))
  #cont += 1
  if n == 999:
    break
  s += n
#Gambiarra
#s -= 999
print('A soma vale {}'.format(s))

# Actualização do python, versão 3.6.3
# Python enhancement proposal 4.9.8
# f-strings, simplificam esta forma de ...

print(f'A soma vale {s}') # Interpolação dentro de strings, já está aceitando esta PEP f-string (actualizado)

Digite um número: 90
Digite um número: 20
Digite um número: 999
A soma vale 110
A soma vale 110


In [10]:
nome = 'José'
idade = 33
print(f'O {nome} tem {idade} anos.')  # PYTHON 3.6+
print('O {} tem {} anos.'.format(nome, idade))  # PYTHON 3
print('O %s tem %d anos.' % (nome, idade)) # PYTHON 2

O José tem 33 anos.
O José tem 33 anos.
O José tem 33 anos.


In [16]:
nome = 'José'
idade = 33
salário = 987.35
print(f'O {nome:-^20} tem {idade} anos e ganha R${salário}')#:.2f}') # < alinhado à esquerda!

O --------José-------- tem 33 anos e ganha R$987.35


# Desafio 66

Crie um programa que leia **vários números** inteiros pelo teclado. O programa só vai parar quando o usuário digitar o valor **999**, que é a **condição de parada**. No final, mostre **quantos números** foram digitados e qual foi a **soma** entre eles (desconsiderando o flag).

In [18]:
count = acc = 0
n = float(input(' Digit the value: '))
while True: # n != 999
  count +=1
  acc += n
  n = float(input('Digit the value: '))
  if n == 999:
    break
print(f'There was {count} numbers read and the sum of them is {acc}')

 Digit the value: 32
Digit the value: 21
Digit the value: 999
There was 2 numbers read and the sum of them is 53.0


In [21]:
count = acc = 0
n = float(input(' Digit the value: '))
while n != 999:
  count +=1
  acc += n
  n = float(input('Digit the value: '))
print(f'There was {count} numbers read and the sum of them is {acc}')

 Digit the value: 1.2
Digit the value: 3.4
Digit the value: 999
There was 2 numbers read and the sum of them is 4.6


In [2]:
# Solução Gambiarra do Gustavo Guanabara
num = soma = 0
while num != 999:
  num = int(input('Digite um valor (999 para parar): '))
  soma += num
soma -= 999
print(f'A soma dos valores foi {soma}!')

Digite um valor (999 para parar): 321
Digite um valor (999 para parar): 98
Digite um valor (999 para parar): 999
A soma dos valores foi 419!


In [3]:
# Solução Gustavo Guanabara
soma = cont = 0
while True:
  num = int(input('Digite um valor (999 para parar): '))
  if num == 999:
    break
  cont += 1
  soma += num
print(f'A soma dos valores foi {soma}!')

Digite um valor (999 para parar): 523
Digite um valor (999 para parar): 87
Digite um valor (999 para parar): 21
Digite um valor (999 para parar): 999
A soma dos valores foi 631!


# Desafio 067

Faça um programa que mostre a **tabuada** de **vários números**, um de cada vez, para cada valor digitado pelo usuário. O programa será **interrompido** quando o número solicitado for **negativo**.

In [44]:
n = ''
print(f"{n:-^20} TABUADA'(S) {n:-^20}\n")

tab = int(input(' What your number choice? ')) 
i=0
while tab >= 0:
  while i < 11:
#while tab >= 0 and i<11: # Bonito se quiser somente uma tab
    print(f'{tab} * {i} = {tab*i}')
    i += 1
  tab = int(input(' What is next multiplication table? '))
  i = 0
print('\nWhen a number is negative, we finish the program ... ')


-------------------- TABUADA'(S) --------------------

 What your number choice? -1

When a number is negative, we finish the program ... 


In [45]:
n = ''
print(f"{n:-^20} TABUADA'(S) {n:-^20}\n")

tab = int(input(' What your number choice? ')) 
i=0
while True:
  while i < 11:
#while tab >= 0 and i<11: # Bonito se quiser somente uma tab
    print(f'{tab} * {i} = {tab*i}')
    i += 1
  tab = int(input(' What is next multiplication table? '))
  i = 0
  if tab < 0:
    break
print('\nWhen a number is negative, we finish the program ... ')

-------------------- TABUADA'(S) --------------------

 What your number choice? 78
78 * 0 = 0
78 * 1 = 78
78 * 2 = 156
78 * 3 = 234
78 * 4 = 312
78 * 5 = 390
78 * 6 = 468
78 * 7 = 546
78 * 8 = 624
78 * 9 = 702
78 * 10 = 780
 What is next multiplication table? 23
23 * 0 = 0
23 * 1 = 23
23 * 2 = 46
23 * 3 = 69
23 * 4 = 92
23 * 5 = 115
23 * 6 = 138
23 * 7 = 161
23 * 8 = 184
23 * 9 = 207
23 * 10 = 230
 What is next multiplication table? -0
0 * 0 = 0
0 * 1 = 0
0 * 2 = 0
0 * 3 = 0
0 * 4 = 0
0 * 5 = 0
0 * 6 = 0
0 * 7 = 0
0 * 8 = 0
0 * 9 = 0
0 * 10 = 0
 What is next multiplication table? -1

When a number is negative, we finish the program ... 


In [6]:
# Solution by Gustavo Guanabara

while True:
  n = int(input('Quer ver a tabuada de qual valor? ')) # ele fez no início e assim poupo linha de código fora do ciclo
  print('-' * 30)
  if n < 0:
    break
  for c in range(1,11):
    print(f'{n} * {c} = {n*c}')
  print('-' * 30)
print('PROGRAMA TABUADA ENCERRADO. Volte Sempre!')


Quer ver a tabuada de qual valor? 23
------------------------------
23 * 1 = 23
23 * 2 = 46
23 * 3 = 69
23 * 4 = 92
23 * 5 = 115
23 * 6 = 138
23 * 7 = 161
23 * 8 = 184
23 * 9 = 207
23 * 10 = 230
------------------------------
Quer ver a tabuada de qual valor? -2
------------------------------
PROGRAMA TABUADA ENCERRADO. Volte Sempre!


# Desafio 068

Faça um programa que jogue **par ou ímpar** com o computador. O jogo só será interrompido quando o jogador **PERDER**, mostrando o total de vitórias consecutivas que ele conquistou no final do jogo.

In [69]:
from random import randint
import time
ganas = 0 
while True:
  pc = randint(0,999) # Fica enviesado ... 0 a 998 Not, ele inclui o 999 então o números de pares e ímpares é igual!
  you = int(input('Say a number of interval [0 to 999]: '))
  you2 = int(input('Achas que na soma vai dar Ímpar?\n[Par] <- No \n[Ímpar] <- Yes\n '))
  print(f'\nSabendo a resposta ... {pc + you} =')
  res = (pc + you) % 2
  if res:
    print('Ímpar') # 1
  else: 
    print('Par') # 0
  time.sleep(2.5)
  if res == you2%2:
    print('You WIN!\n ')
    ganas += 1
    print(f'Round {ganas+1})
  else:
    print('You lose ... \n Program finnishing ... \n')
    print(f'You win {ganas:8^} times')
    break


Say a number of interval [0 to 999]: 76
Achas que na soma vai dar Ímpar?
[Par] <- No 
[Ímpar] <- Yes 8

Sabendo a resposta ... 852 =
Par
You WIN!
Say a number of interval [0 to 999]: 69
Achas que na soma vai dar Ímpar?
[Par] <- No 
[Ímpar] <- Yes 0

Sabendo a resposta ... 886 =
Par
You WIN!
Say a number of interval [0 to 999]: 999
Achas que na soma vai dar Ímpar?
[Par] <- No 
[Ímpar] <- Yes 3

Sabendo a resposta ... 1163 =
Ímpar
You WIN!
Say a number of interval [0 to 999]: 76
Achas que na soma vai dar Ímpar?
[Par] <- No 
[Ímpar] <- Yes 0

Sabendo a resposta ... 210 =
Par
You WIN!
Say a number of interval [0 to 999]: 99
Achas que na soma vai dar Ímpar?
[Par] <- No 
[Ímpar] <- Yes 0

Sabendo a resposta ... 860 =
Par
You WIN!
Say a number of interval [0 to 999]: 77
Achas que na soma vai dar Ímpar?
[Par] <- No 
[Ímpar] <- Yes 1

Sabendo a resposta ... 93 =
Ímpar
You WIN!
Say a number of interval [0 to 999]: 12
Achas que na soma vai dar Ímpar?
[Par] <- No 
[Ímpar] <- Yes 0

Sabendo a respo

In [86]:
from random import randint
import time
ganas = 0 
print(f'Round {ganas+1:_<20}\n ')

while True: # Fiz tudo num único while e o que ele fez foi..: foi arranjar mais um while para P or Í que garante que é letra se der número manda repetir. O meu tem de ser número.., se for letra F****** 
  pc = randint(0,999)
  you = int(input('Say a number of interval [0 to 999]: '))
  you2 = int(input('Achas que na soma vai dar Ímpar?\n[Par] <- No \n[Ímpar] <- Yes\n '))
  print(f'\n... Sabendo a resposta ... {pc + you} =')
  res = (pc + you) % 2
  if res:
    print('Ímpar') # 1
  else: 
    print('Par') # 0
  time.sleep(2.5)
  if res == you2%2:
    print('      You WIN!\n\n ')
    ganas += 1
    print(f'Round {ganas+1:-<20}')
  else:
    print('You lose ... \n Program finnishing ... \n')
    print(f'You win {ganas:8^} times')
    break


Round 1___________________
 
Say a number of interval [0 to 999]: 34
Achas que na soma vai dar Ímpar?
[Par] <- No 
[Ímpar] <- Yes
 2

... Sabendo a resposta ... 366 =
Par
      You WIN!

 
Round 2-------------------
Say a number of interval [0 to 999]: 889
Achas que na soma vai dar Ímpar?
[Par] <- No 
[Ímpar] <- Yes
 1

... Sabendo a resposta ... 1181 =
Ímpar
      You WIN!

 
Round 3-------------------
Say a number of interval [0 to 999]: 56
Achas que na soma vai dar Ímpar?
[Par] <- No 
[Ímpar] <- Yes
 1

... Sabendo a resposta ... 105 =
Ímpar
      You WIN!

 
Round 4-------------------
Say a number of interval [0 to 999]: 76
Achas que na soma vai dar Ímpar?
[Par] <- No 
[Ímpar] <- Yes
 3

... Sabendo a resposta ... 1061 =
Ímpar
      You WIN!

 
Round 5-------------------
Say a number of interval [0 to 999]: 87
Achas que na soma vai dar Ímpar?
[Par] <- No 
[Ímpar] <- Yes
 2

... Sabendo a resposta ... 935 =
Ímpar
You lose ... 
 Program finnishing ... 

You win 4 times


In [15]:
# Solution by Gustavo Guanabara

from random import randint
v = 0
while True:
  jogador = int(input('Diga um valor: '))
  computador = randint(0, 11)
  total = jogador + computador
  tipo = ' '
  while tipo not in 'PI': #'PpIi':
    tipo = str(input('Par ou Ímpar? [P/I] ')).strip().upper()[0]
  print(f'Voçê jogou {jogador} e o computador {computador}. Total de {total}')
  print('DEU PAR' if total % 2 == 0 else 'DEU ÍMPAR')
  if tipo == 'P':
    if total % 2 == 0:
      print('Voçê VENCEU!')
      v += 1
    else:
      print('Voçê PERDEU')
      break
  elif tipo == 'I':
    if total % 2 == 1:
      print('Voçê VENCEU!')
      v += 1
    else:
      print('Voçê PERDEU')
      break
  print('Vamos jogar novamente...')
print(f'GAME OVER! Voçê venceu {v} vez(es). ')
    



Diga um valor: 54
Par ou Ímpar? [P/I] 4
Par ou Ímpar? [P/I] 3
Par ou Ímpar? [P/I] 2
Par ou Ímpar? [P/I] 08
Par ou Ímpar? [P/I] p
Voçê jogou 54 e o computador 4. Total de 58
DEU PAR
Voçê VENCEU!
Vamos jogar novamente...
Diga um valor: 43
Par ou Ímpar? [P/I] f
Par ou Ímpar? [P/I] I
Voçê jogou 43 e o computador 9. Total de 52
DEU PAR
Voçê PERDEU
GAME OVER! Voçê venceu 1 vez(es). 


In [12]:
from random import randint
randint(0,1)

0

# Desafio 069 

Crie um programa que leia a **idade** e o **sexo** de **várias pessoas**. A cada pessoa cadastrada, o programa deverá perguntar se o **usuário** quer ou não continuar.
No final, mostre:

**A)** quantas pessoas tem mais de **18 anos**.
**B)** Quantos **homens** façam cadastrados.
**C)** Quantas **mulheres** tem menos de **20 anos.**

In [97]:
m18 = men = girls = 0
continuing = 77
while continuing:
  idade = int(input('Please, say your age: '))
  while idade < 0:
    idade = int(input('Please, say your age: '))
  sexo = str(input('Please, say your gender: ')).strip().upper()
  while sexo not in 'FM':
    sexo = str(input('Please, say your gender: ')).strip().upper()
  if idade >= 18:
    m18 += 1
  if sexo == 'M':
    men += 1
  elif idade <= 20: # +Bem aproveitado face ao do Gustavo , tlv menos legível... 
    girls += 1
  continuing = int(input('Do you wish continue? \n[NO] <- 0\n[YES] <- Number ≠ 0\t:')) # quer continuar com not in -... okay e depois as escolhas if S
print(f'There are:\n - {m18} adults\n - {men} mens\n - {girls} girls\n')
  #if sexo
  # muito importante cadastrar em base de dados e tudo mais.  CLásssico, estatística com dados e fazer alg. contas!

Please, say your age: 81
Please, say your gender: M
Do you wish continue? 
[NO] <- 0
[YES] <- Number ≠ 0	:5
Please, say your age: 98
Please, say your gender: f
Do you wish continue? 
[NO] <- 0
[YES] <- Number ≠ 0	:3
Please, say your age: 12
Please, say your gender: f
Do you wish continue? 
[NO] <- 0
[YES] <- Number ≠ 0	:0
There are:
 - 2 adults
 - 1 mens
 - 1 girls



In [20]:
# Solution by Gustavo Guanabara , estatistica sobre dados
tot18 = totH = totM20 = 0
while True:
  idade = int(input('Idade: '))
  sexo = ' '
  while sexo not in 'MF':
    sexo = str(input('Sexo: [M/F]')).strip().upper()[0]
  if idade >= 18: # fica a aceita idades negativas?
  
    tot18 += 1
  if sexo == 'M':
    totH += 1
  if sexo == 'F' and idade < 20:
    totM20 += 1
  resp = ' '
  while resp not in 'SN':
    resp = str(input('Quer continuar? [S/N] ')).strip().upper()[0]
  if resp == 'N': 
    break # não usei break, ao invês coloquei o 0
print(f'Total de pessoas com mais de 18 anos: {tot18}')
print(f'Ao todo temos {totH} homens cadastrados')
print(f'E temos {totM20} mulheres com menos de 20 anos')

Idade: -19
Sexo: [M/F]f
Quer continuar? [S/N] s
Idade: -7
Sexo: [M/F]M
Quer continuar? [S/N] n
Total de pessoas com mais de 18 anos: 0
Ao todo temos 1 homens cadastrados
E temos 1 mulheres com menos de 20 anos


# Desafio 70

Crie um programa que leia o **nome** e o **preço** de **vários produtos**. O programa deverá perguntar se o **usuário** vai continuar. No final, mostre:

**A)** Qual é o **total gasto** na compra.

**B)** Quantos produtos custam **mais** de **R$1000**.

**C)** Qual é o **nome** do produto mais **barato**. 



In [33]:
tot = exp = 0
price = float(input(' Product price: R$'))#Say the product price: '))
name = str(input(' Please, product name: \n'))#, say product name: \n'))
cheap = price
cheapest = name #1 ele colocou tudo junto, em troca de acrescentar um contador! para o primeiro product!!
continuing = 'Y'
while continuing in 'YN':
  tot += price
  if price > 1000:
    exp += 1
  if price < cheap:
    cheap = price
    cheapest = name
  continuing = str(input('\n Continuing Yes or No ')).strip().upper()[0]
  while continuing not in 'YN':
    continuing = str(input(' Continuing Yes or No ')).strip().upper()[0]
  if continuing == 'N':
      break
  price = float(input('\n\n Product price: R$'))
  name = str(input(' Please, product name: \n'))
print(f' ___________________ END ______________________\nThe total cost is {tot}\n There are {exp} products with cost > R$1000 ! \n The product more cheaper is {cheapest}!')

 Product price: R$89
 Please, product name: 
cARACOL

 Continuing Yes or No nocan
 ___________________ END ______________________The total cost is 89.0
 There are 0 products with cost > R$1000 ! 
 The product more cheaper is cARACOL!


In [98]:
'Yes'.strip().upper()[0]

'Y'

In [101]:
' yes Motherfucker ! '.strip().upper()[0]

'Y'

In [35]:
# Solution by Gustavo Guanabara  Mto bom 2 dicas interessantes, esperteza!
total = totmil = cont = menor = 0 # qual é o produto mais barato e depois o nome dele
while True: 
  produto = str(input('Nome do Produto: '))
  preço = float(input('Preço: R$ '))
  cont += 1
  total += preço
  if preço > 1000:
    totmil += 1
  if cont == 1 or preço < menor: # NÃO COLOCAR gambiarra de menor_preço = 999, o primeiro coloca sempre o min = 1º produto! .. é um do, do while (na prática)
    menor = preço
    barato = produto
  # else:
  #   if preço < menor:
  #     menor = preço
  #     barato = produto
  resp = ' ' #2 como ele obriga a mudar e a entrar no ciclo que depois impõe a (formato) começar em Y ou N..
  while resp not in 'SN':
    resp = str(input('Quer continuar? [S/N] ')).strip().upper()[0]
  if resp == 'N':
    break
print('{:-^40}'.format(' FIM DO PROGRAMA '))
#print(' FIM DO PROGRAMA ')
print(f'O total da compra foi R${total:.2f}')
print(f'Temos {totmil} produtos custando mais de R$1000.00')
print(f'O produto mais barato foi {barato} que custa R${menor:.2f}')


Nome do Produto: Caju
Preço: R$33
Quer continuar? [S/N] n
----------- FIM DO PROGRAMA ------------
O total da compra foi R$33.00
Temos 0 produtos custando mais de R$1000.00
O produto mais barato foi Caju que custa R$33.00


# Desafio 071

Crie um programa que simule o funcionamento de um **caixa electrônico**. No início, pergunte ao usuário qual será o **valor a ser sacado** (número inteiro) e o programa vai informar quantas **cédulas** de cada valor serão entregues.

**OBS:** Considere que o caixa possui cédulas de R$50, R$20 , R$10 e R$1.

In [123]:
ast = '*'
print(f'{ast*20} SIBS ATM ********************* \n\n')
value = int(input(' Please, how much money do you need: '))
b50 = b20 = b10 = c1 = 0  # ninguém saca centavos da caixa electronico
money = value
while money > 0: 
  # dividir \ sera mais fácil... divisão inteira
  while money > 50:
    money -= 50
    b50 += 1
  while money > 20: # se pusesse encadeado, mais demorado... para que complicar ...
    money -= 20 
    b20 += 1
  while money > 10: # and b10A > 0 ... pensando que haverá sempre notas
    money -= 10
    b10 += 1
  while money > 0:
    money -= 1
    c1 += 1
  print(f'{value} = \n{b50} R$50\n{b20} R$20\n{b10} R$10\n{c1} R$1')
print(f'{ast*10} Volte Sempre! {ast*10}')

******************** SIBS ATM ********************* 


 Please, how much money do you need: 342
342 = 
6 R$50
2 R$20
0 R$10
2 R$1
********** Volte Sempre! **********


In [127]:
money = value # solução minha inteligente!

while money > 0:

  b50 = money // 50 #/ 50
  money = money % 50

  b20 = money // 20
  money = money % 20

  b10 = money // 10
  money = money % 10

  c1 = money // 1
  money -= c1
  if money == 0:
    print(f'{value} = \n{b50} R$50\n{b20} R$20\n{b10} R$10\n{c1} R$1')
  else:
    print(f'Nota de crédito {money}, sem numerário para si! ')
  break
print(f'{ast*10} Volte Sempre! {ast*10}') # negativos, digo : esta máquina não aceita depósitos

342 = 
6 R$50
2 R$20
0 R$10
2 R$1
********** Volte Sempre! **********


In [39]:
print('*' * 30)
print('{:^30}'.format('Caixa ATM'))
print('*' * 30)
montante = int(input(' Qtd dinheiro: '))
nota = 50
while montante > 0:
  qtd = montante // nota
  print(f'Total de {qtd} cédulas de R${nota}')
  montante = montante % nota
  if nota == 50: # 50 -> 20 -> 10 -> 1
    nota = 20
  elif nota == 20:
    nota = 10
  elif nota == 10:
    nota = 1
  else:
    break
print('{:^30}'.format('Operação chegou ao Fim, Volte Sempre!'))


******************************
          Caixa ATM           
******************************
 Qtd dinheiro: 238
Total de 4 cédulas de R$50
Total de 1 cédulas de R$20
Total de 1 cédulas de R$10
Total de 8 cédulas de R$1
Operação chegou ao Fim, Volte Sempre!


In [36]:
# Solution by Gustavo Guanabara  [ consistente e flexível] dá para acrescentar mais notas etc

print('=' * 30)
print('{:^30}'.format('BANCO CEV'))
print('=' * 30)
valor = int(input('Que valor voçê quer sacar? R$'))
total = valor
céd = 50
totcéd = 0
while True:
  if total >= céd:
      total -= céd
      totcéd += 1
  else:
    if totcéd > 0:
      print(f'Total de {totcéd} cédulas de R${céd}')
    if céd == 50: # está sempre a fazer intercâmbio do valor da variável. ... e depois com a imprime? é um ciclo de impressão com valores diferentes
        céd = 20
    elif céd == 20:
        céd = 10
    elif céd == 10:
      céd = 1
    totcéd = 0
    if total == 0:
      break
print('=' * 30)
print('Volte sempre ao BANCO CEV! Tenha um bom dia!')

          BANCO CEV           
Que valor voçê quer sacar? R$125
Total de 2 cédulas de R$50
Total de 1 cédulas de R$20
Total de 5 cédulas de R$1


In [40]:
n = "Ola"
m = "key"

In [43]:
n[0] in 'aeiou'.upper()

True

In [1]:
'''variável composta: tuplas, lista, dicionário

strings, fatiamento... dentro do pyython tmb funciona ..., 

colchetes, durante o fatiamento ... divisão


lanche -1 começa no fim!
estruturas de repetição, ...

iteração da estrutura


for com o range e para coleções (variáveis compostas como a tupla).

termo : as tuplas são imutáveis... não é possível trocar (tirar o pudim da tuplas. fica assim durante o programa diferente!)'''


'variável composta: tuplas, lista, dicionário\n\nstrings, fatiamento... dentro do pyython tmb funciona ..., \n\ncolchetes, durante o fatiamento ... divisão\n\n\nlanche -1 começa no fim!\nestruturas de repetição, ...\n\niteração da estrutura\n\n\nfor com o range e para coleções (variáveis compostas como a tupla).\n\ntermo : as tuplas são imutáveis... não é possível trocar (tirar o pudim da tuplas. fica assim durante o programa diferente!)'

In [4]:
'''
3 tipos de variáveis compostas:
tuple      <- ()  # entender cada uma delas (o que pode ser feito com elas) e as diferenças entre as v.c
list       <- [] # colchete
dictionary <- {}
'''

lanche = 2 # variável, é um espaço na memória, simples
lanche = 3 # substitui o que estava na memória

lanche = (1,3,2,4) # 3 possibilidades de armazenar 4 valores numa variável (composta, em forma de tuple, list or dictionary)

# fatiamento, como aconteceu com as strings

# analogia as strings, que tmb já são variáveis compostas (manipulação), funcionar antes da aula...
lanche[0] # elementos diferentes, como posso acessar a esses elementos da tupla ...? através dos índices
# manipulação das strings, através do fatiamento!

1

In [8]:
lanche
lanche[0:2] # exclui o index 2 que é o 3º elemento
lanche[1:] # 1: écomece no 1 e vai até ao final, fatiamento simples
lanche[-1] # último elemento (começa no final quando tem menos)
len(lanche) # size, repetições while, while TRUE infinito + break para interromper, for

for c in lanche: # implica a iteração, em relação ao C
  print(c) # for utilizando com o método range bem como em tupla etc. este 4 são os índexes 0,1,2,3

1
3
2
4


In [22]:
# tuplas são imutáveis, não conseguimos mudar a tupla ... só correndo de novo o programa

#lanche = 'Hambúrguer'
#lanche = 'Suco'

# lanche = ('Hambúrguer') # a partir do python 3.5 nem precisamosde colocar parentisis ... ora teste na linha abaixo
lanche = 'Hambúrguer', 'Suco', 'Pizza', 'Pudim'
lanche = ('Hambúrguer', 'Suco', 'Pizza', 'Pudim')
print(lanche[1])
print(lanche[-2])
print(lanche[1:3]) # desconsidera o último elemento, mostra um pedaço da tupla
print(lanche[2:])
print(lanche[:2]) # mostra os elementos c/ indexes 0 e 1
print(lanche[-4:-2]) # pensar na vírgulas com interregnios , contar as vírgulas como se fizessem barreiras e as últimas são os próprios colchetes. Isto mais útil quando são : pedaços, mas ele por norma segue para a direita -2: conta 2 vírgulas a contar do final e depois vai até ao final (mesmo caminho.); se for :-2 é do início e depois começa a contar do fim até 2 virgulas e highlited o que está no meio!
print(lanche[-2:])

Suco
Pizza
('Suco', 'Pizza')
('Pizza', 'Pudim')
('Hambúrguer', 'Suco')
('Hambúrguer', 'Suco')
('Pizza', 'Pudim')


In [32]:
lanche[-3:]

('Suco', 'Pizza', 'Pudim')

In [29]:
lanche[:-2]
lanche[:-3] # mostra a vírfula, que interessante!

('Hambúrguer',)

In [24]:
print(lanche[0:5]) # funciona igual ..

('Hambúrguer', 'Suco', 'Pizza', 'Pudim')


In [41]:
# Tuplas são imutáveis

# lanche[1] = 'Refrigerante' # TypeError: 'tuple' object does not support item assignment
print(lanche[1])

for comida in lanche:
  print(f'Eu vou comer {comida}')#lanche[1]}')
print('Comi pra Caramba!')

Suco
Eu vou comer Hambúrguer
Eu vou comer Suco
Eu vou comer Pizza
Eu vou comer Pudim
Comi pra Caramba!


In [43]:
lanche = ('Hambúrguer', 'Suco', 'Batata Frita', 'Pizza', 'Pudim')

for cont in range(0, len(lanche)):
  print(f' Vou comer {lanche[cont]} na posição {cont}')  

 Vou comer Hambúrguer na posição 0
 Vou comer Suco na posição 1
 Vou comer Batata Frita na posição 2
 Vou comer Pizza na posição 3
 Vou comer Pudim na posição 4


In [6]:
del index, element

SyntaxError: ignored

In [9]:
lanche = ('Hambúrguer', 'Suco', 'Batata Frita', 'Pizza', 'Pudim')

for index, element in enumerate(lanche):
  print(f' Vou comer {element} na posição {index}')  # enumerate retorna ... 

 Vou comer Hambúrguer na posição 0
 Vou comer Suco na posição 1
 Vou comer Batata Frita na posição 2
 Vou comer Pizza na posição 3
 Vou comer Pudim na posição 4


In [10]:
for pos, comida in enumerate(lanche):
  print(f'Eu vou comer {comida} na posição {pos}')

Eu vou comer Hambúrguer na posição 0
Eu vou comer Suco na posição 1
Eu vou comer Batata Frita na posição 2
Eu vou comer Pizza na posição 3
Eu vou comer Pudim na posição 4


In [47]:
name_list = ['Jane', 'John', 'Mike']
list(enumerate(name_list))

[(0, 'Jane'), (1, 'John'), (2, 'Mike')]

In [56]:
name_list = ('Jane', 'John', 'Mike')
list(enumerate(name_list))

[(0, 'Jane'), (1, 'John'), (2, 'Mike')]

In [5]:
name = "Eric"
profession = "comedian"
affiliation = "Monty Python"
message = (
  f"Hi {name}. "
  f"You are a {profession}. "
  f"You were in {affiliation}."
)
message

'Hi Eric. You are a comedian. You were in Monty Python.'

In [12]:
print(lanche)
print(sorted(lanche)) # #não mudou#, apenas colocou em ordem! não altera simplestemente coloca em ordem
print(lanche) # estado natural

('Hambúrguer', 'Suco', 'Batata Frita', 'Pizza', 'Pudim')
['Batata Frita', 'Hambúrguer', 'Pizza', 'Pudim', 'Suco']


In [23]:
a = (2,5,4)
b = (5,8,1,2)
print(a)
print(b)
c = b + a
print(c) # junta as tuplas, a ordem tem total influência! Mais uma funcionalidade aí!
print(len(c))

print(c.count(5)) # o cinco aparece duas vezes
print(c.count(9))

print(c.index(8))# em que posição está o 8
print(c.index(5))# vai buscar o local da 1ª ocurrência

print(c.index(5,1))# se for buscar após a 1ª posição, segundo argumento da função, deslocamento segundo argumento

(2, 5, 4)
(5, 8, 1, 2)
(5, 8, 1, 2, 2, 5, 4)
7
2
0
1
0
5


In [27]:
pessoa = (1,2,3)

pessoa = ('Gustavo', 39, 'M', 99.88) # aceita vários tipos de dados, em outras linguagens não (na generalidade)/ dados de tipos diferentes na tupla
print(pessoa)

#del(pessoa[0]) #TypeError: 'tuple' object doesn't support item deletion

('Gustavo', 39, 'M', 99.88)


TypeError: ignored

In [26]:
del (pessoa)# apagar uma variável
#pessoa # NameError: name 'pessoa' is not defined



NameError: ignored

# Desafio 072

Crie um programa que tenha uma **tupla** totalmente prenchida com uma contagem por extenso, de **zero** até **vinte**.

Seu programa deverá ler um número pelo teclado (entre 0 e 20) e mostrá-lo por **extenso**.

In [32]:
tupla = ('zero', 'um', 'dois', 'três', 'quatro', 'cinco', 'seis', 'sete', 'oito', 'nove', 'dez', 'onze', 'doze', 'treze', 'catorze', 'quinze', 'dezesseis', 'dezassete', 'dezoito', 'dezanove', 'vinte')

for seq in range(0,21):
  print(tupla[seq])

zero
um
dois
três
quatro
cinco
seis
sete
oito
nove
dez
onze
doze
treze
catorze
quinze
dezesseis
dezassete
dezoito
dezanove
vinte


In [39]:
userNum = int(input('Please, say a number in interval between [0,20]: '))
while userNum < 0 or userNum > 20:
  userNum = int(input('Please try again!\n\n Say a number in interval between [0,20]: '))
for seq in tupla:
  print(f'You choose {tupla[userNum]}')
  break

Please, say a number in interval between [0,20]: 6
You choose seis


# Desafio 073

Crie uma **tupla** preenchida com os **20 primeiros** colocados da Tabela do **Campeonato Brasileiro de Futebol**, na ordem de colocação. Depois mostre:

A) Apenas os **5** **primeiros** colocados.

B) Os **últimos 4** colocados da tabela.

C) Uma lista com os times em **ordem alfabética**.

D) Em que **posição** na tabela está o time da **Chapecoense**.


In [ ]:
tupla = ('Palmeiras', 'Corinthians', 'Fluminense', 'Athletico-PR', 'Flamengo', 'Internacional', 'Atlético-MG', 'Bragantino', 'Santos', 'São Paulo', 'Goías', 'Botafogo', 'América-MG', 'Ceará SC', 'Coritiba', 'Avaí', 'Cuiabá', 'Fortaleza', 'Atlético-GO', 'Juventude'  )


#A
best = 5

for clubs in enumerate()

# Como não há Chapecoense, emito a mensagem que não está no Brasileirão e por exemplo a equipa tal tal está no lugar tal